File: 0717-1-bit-and-2-bit-characters/0717-1-bit-and-2-bit-characters.java
class Solution {
    public boolean isOneBitCharacter(int[] bits) {

        int count = 0;
        int index = 0;
        int length = bits.length;

        while(index < length)
        {
            if(bits[index] == 0)
            {
                index++;
                count = 1;
            }
            else
            {
                index +=2;
                count = 2;  
            }
        }   

        return count == 1 ? true : false;
    }
}

File: 0717-1-bit-and-2-bit-characters/0717-1-bit-and-2-bit-characters.js
/**
 * @param {number[]} bits
 * @return {boolean}
 */
var isOneBitCharacter = function(bits) {
    
    let length = bits.length;
    let index = 0;
    let ans = 0;

    while(index < length)
    {
        if(bits[index] == 0)
        {
            ans = 1;
            index++;
        }
        else
        {
            ans = 2;
            index += 2;
        }
    }

    return ans == 1 ? true : false;
};